pkg/storage/stores/shipper/downloads/table_manager_test.go

```diff
@@ -35,29 +36,50 @@ func buildTestTableManager(t *testing.T, path string) (*TableManager, stopFunc)
 }
 
 func TestTableManager_QueryPages(t *testing.T) {
-	tempDir := t.TempDir()
-
-	objectStoragePath := filepath.Join(tempDir, objectsStorageDirName)
+	t.Run("QueryPages", func(t *testing.T) {
+		tempDir := t.TempDir()
+		objectStoragePath := filepath.Join(tempDir, objectsStorageDirName)
 
-	var queries []chunk.IndexQuery
-	for i, name := range []string{"table1", "table2"} {
-		testutil.SetupTable(t, filepath.Join(objectStoragePath, name), testutil.DBsConfig{
-			NumUnCompactedDBs: 5,
-			DBRecordsStart:    i * 1000,
-		}, testutil.PerUserDBsConfig{
-			DBsConfig: testutil.DBsConfig{
+		var queries []chunk.IndexQuery
+		for i, name := range []string{"table1", "table2"} {
+			testutil.SetupTable(t, filepath.Join(objectStoragePath, name), testutil.DBsConfig{
 				NumUnCompactedDBs: 5,
-				DBRecordsStart:    i*1000 + 500,
-			},
-			NumUsers: 1,
-		})
-		queries = append(queries, chunk.IndexQuery{TableName: name})
-	}
+				DBRecordsStart:    i * 1000,
+			}, testutil.PerUserDBsConfig{
+				DBsConfig: testutil.DBsConfig{
+					NumUnCompactedDBs: 5,
+					DBRecordsStart:    i*1000 + 500,
+				},
+				NumUsers: 1,
+			})
+			queries = append(queries, chunk.IndexQuery{TableName: name})
+		}
 
-	tableManager, stopFunc := buildTestTableManager(t, tempDir)
-	defer stopFunc()
+		tableManager, stopFunc := buildTestTableManager(t, tempDir)
+		defer stopFunc()
 
-	testutil.TestMultiTableQuery(t, testutil.BuildUserID(0), queries, tableManager, 0, 2000)
+		testutil.TestMultiTableQuery(t, testutil.BuildUserID(0), queries, tableManager, 0, 2000)
+	})
+
+	t.Run("it doesn't deadlock when table create fails", func(t *testing.T) {
+		tempDir := os.TempDir()
```

Comment:
Just a nit, lets use `t.TempDir` like other places.

Reply:
Sorry, I pushed this change and resolved the merge conflict because I wanted to cut a new release with this fix.